packages/lit-dev-tools-cjs/src/api-docs/generate.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as typedoc from 'typedoc';
+import * as fs from 'fs/promises';
+import * as pathlib from 'path';
+import {execFile} from 'child_process';
+import {promisify} from 'util';
+import {ApiDocsTransformer} from './transformer.js';
+import {lit2Config} from './configs/lit-2.js';
+
+import type {ApiDocsConfig} from './types.js';
+
+const execFileAsync = promisify(execFile);
+
+const configs = [lit2Config];
+
+/**
+ * Check whether the given file path exists.
+ */
+const fileExists = async (path: string): Promise<boolean> => {
+  try {
+    await fs.stat(path);
+    return true;
+  } catch (err) {
+    if ((err as {code?: string}).code !== 'ENOENT') {
+      throw err;
+    }
+  }
+  return false;
+};
+
+/**
+ * Clone the given Git repo URL at the given commit SHA into the given
+ * directory. If the directory already exists, do nothing.
+ */
+const cloneIfNeeded = async (repo: string, sha: string, dir: string) => {
+  if (await fileExists(dir)) {
+    console.log(`${dir} already exists, skipping git clone`);
```

Comment:
shouldn't you make sure to do the checkout in case the directory exists, but not at the right SHA?

Reply:
I want to support the use case where you are working on the code annotations directly from the cloned directory. If you make a change and then check that in, it would be bad to automatically blow that change away when you then go and re-generate the docs.

I added a check that the SHAs match, and throws an exception if it doesn't, telling you to either update the config or delete the directory to fix things.